tools/lsp/server_loop.rs

```diff
@@ -107,6 +107,18 @@ impl DocumentCache {
         );
         Some(pos)
     }
+
+    pub fn mapper(&mut self, uri: lsp_types::Url) -> OffsetToPositionMapper {
```

Comment:
mapper is a bit vague on the DocumentCache.  mapper of what.  I suggest

```suggestion
    pub fn offset_to_position_mapper(&mut self, uri: lsp_types::Url) -> OffsetToPositionMapper {
```